ui/app/src/components/CompareVersionsDialog/RepeatGroupItems.tsx

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright (C) 2007-2024 Crafter Software Corporation. All Rights Reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as published by
+ * the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import ContentInstance from '../../models/ContentInstance';
+import { ContentTypeField, LookupTable } from '../../models';
+import React, { useEffect, useState } from 'react';
+import useSpreadState from '../../hooks/useSpreadState';
+import { fromString, serialize } from '../../utils/xml';
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import ArrowBackIosRounded from '@mui/icons-material/ArrowBackIosRounded';
+import { FormattedMessage } from 'react-intl';
+import CompareArrowsRoundedIcon from '@mui/icons-material/CompareArrowsRounded';
+import palette from '../../styles/palette';
+import Typography from '@mui/material/Typography';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import Checkbox from '@mui/material/Checkbox';
+import { CompareFieldPanel } from './CompareVersions';
+import { areObjectsEqual } from '../../utils/object';
+
+interface RepeatGroupItemsProps {
+  contentA: ContentInstance[];
+  contentB: ContentInstance[];
+  aXml: string;
+  bXml: string;
+  fields: LookupTable<ContentTypeField>;
+}
+
+export function RepeatGroupItems(props: RepeatGroupItemsProps) {
+  const { contentA, contentB, aXml, bXml, fields } = props;
+  const [repDiff, setRepDiff] = useState([]);
+  const [repItemsCompare, setRepItemsCompare] = useSpreadState({
+    a: null,
+    b: null
+  });
+  const showRepItemsCompare = repItemsCompare.a?.content && repItemsCompare.b?.content;
+  const selectedItemsAreEqual =
+    showRepItemsCompare && areObjectsEqual(repItemsCompare.a?.content, repItemsCompare.b?.content);
+  const [compareRepItemVersionsMode, setCompareRepItemVersionsMode] = useState(false);
+
+  useEffect(() => {
+    const contentALength = (contentA ?? []).length;
+    const contentBLength = (contentB ?? []).length;
+    const maxLength = contentALength > contentBLength ? contentALength : contentBLength;
+    const diffArray = [];
+
+    for (let x = 0; x < maxLength; x++) {
+      const itemA = contentA?.[x] ? JSON.stringify(contentA[x]) : null;
+      const itemB = contentB?.[x] ? JSON.stringify(contentB[x]) : null;
```

Comment:
JSON.strigify doesn't guarrantee order of properties and overall is likely not a reliable comparison. We already have the XML, it'd make more sense to compare that native string instead, which will also take away the stringification cost.